src/util/defs.h

```diff
@@ -3,26 +3,39 @@
 #include <Qt>
 
 // Maximum buffer length to each EngineObject::process call.
-//TODO: Replace this with mixxx::AudioParameters::bufferSize()
-constexpr unsigned int MAX_BUFFER_LEN = 160000;
+// Note: MAX_BUFFER_LEN shouldn't be use for audio processing,
+// because the buffers should be sized according to the processing
+// block in frames and then multiplied by the channel count.
+// Then audio buffers used in I/O and non realtime operation
+// should be sized according to their task with a balance in mind
+// between CPU efficiency and memory usage.
+//
+// Max frames at 44.1kHz:     4096
+// Max frames at  192kHz: 4 * 4096
+//
+// TODO: Replace this with mixxx::AudioParameters::bufferSize()
+static constexpr unsigned int kMaxProcessFrames = 4 * 4096;
+static constexpr unsigned int kMaxProcessChannels = 2;
+static constexpr unsigned int kMaxProcessSamples = kMaxProcessChannels * kMaxProcessFrames;
+static constexpr unsigned int MAX_BUFFER_LEN = kMaxProcessSamples;
 
-constexpr int kMaxNumberOfDecks = 4;
+static constexpr int kMaxNumberOfDecks = 4;
```

Comment:
We use anonymous namepsaces instead of the almost deprecated static keyword for this purpose.

Reply:
Anonymous namespaces in `.cpp` files serve a different purpose and are not applicable here. The `static` keyword serves many different purposes.

Though I guess `static` is redundant here, because the evaluation of the constant could safely be inlined and the value does not need to be stored at a fixed memory location.

Reply:
good point, though at the same time, the compiler might decide not to actually inline nor evaluate a `constexpr` variable at compile time if you omit the `static`, though this would be incredibly unlikely in the case of an `int`.

Reply:
> Anonymous namespaces in .cpp files serve a different purpose and are not applicable here.

Why do you think that? The compiler has no idea of headers. It is the predecessor that builds a giant Cpp file with all headers literally included. So it is zero differences if the anonymous namespace is defined in the header or the cpp file.

Reply:
I've added static because I wasn't sure if you don't have it if the compiler will have to export a symbol.

Reply:
Here is a good summary of our discussion here:  
https://releases.llvm.org/16.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/misc/use-anonymous-namespace.html

It turns out that `static constexpr` is redundant. Please remove the static keyword.

Reply:
Done.